### DISCUSSION TITLE
Adding an option for process.env vars to be replaced even when context is "node"

### DISCUSSION THREAD

**Discussion by Ricardonacif** [parcel-bundler/parcel#8384] [General] [answered]
For Parcel 1, there was this plugin https://www.npmjs.com/package/parcel-plugin-electron-dotenv?activeTab=readme

I couldn't find anything similar for Parcel 2. I believe its a very common use case, specially for electron applications. I really need help with this, I tried searching for the file where this is done on src but I wasn't able to find. Any tips?

Thanks,

**Comment by mischnic** [comment, marked as answer]
Related code, the first line is the condition for when to replace process.env:
https://github.com/parcel-bundler/parcel/blob/9e2d5d0d60d08d65b5ae6cd765c907a8753bbf39/packages/transformers/js/src/JSTransformer.js#L394

https://github.com/parcel-bundler/parcel/blob/9e2d5d0d60d08d65b5ae6cd765c907a8753bbf39/packages/transformers/js/core/src/lib.rs#L311-L320

https://github.com/parcel-bundler/parcel/blob/9e2d5d0d60d08d65b5ae6cd765c907a8753bbf39/packages/transformers/js/core/src/env_replacer.rs#L24

**Comment by Ricardonacif** [comment]
@mischnic would you guys accept a PR with a new setting to make this optional? If so, any preferences on the API?

**Reply by Ricardonacif** [reply]
ping @mischnic
